cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1983,27 +1983,32 @@ class ObjectOutputStream final : public io::OutputStream {
       const void* data, int64_t nbytes, std::shared_ptr<Buffer> owned_buffer = nullptr) {
     req.SetBucket(ToAwsString(path_.bucket));
     req.SetKey(ToAwsString(path_.key));
-    req.SetBody(std::make_shared<StringViewStream>(data, nbytes));
     req.SetContentLength(nbytes);
     RETURN_NOT_OK(SetSSECustomerKey(&req, sse_customer_key_));
 
     if (!background_writes_) {
-      req.SetBody(std::make_shared<StringViewStream>(data, nbytes));
+      // (GH-45304: avoid setting a body stream if length is 0)
```

Comment:
Could you add a comment that we can remove this workaround when we require aws-sdk-cpp 1.11.489 or later?

Reply:
Ok, done. Thanks for the suggestion.